src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/inbound/InboundUserAcceptanceTest.java

```diff
@@ -48,86 +50,136 @@ void tearDown() {
+
+        final Interchange interchange = getEdifactParser().parse(testData.getEdifact());
+
+        final String recipient = interchange.getInterchangeHeader().getRecipient();
+
+        WorkflowId workflowId;
+        if (testData.getEdifact().contains(WorkflowId.PATHOLOGY.getWorkflowId())) {
+            workflowId = WorkflowId.PATHOLOGY;
+        } else if (testData.getEdifact().contains(WorkflowId.SCREENING.getWorkflowId())) {
+            workflowId = WorkflowId.SCREENING;
+        } else {
+            throw new RuntimeException("Unsupported Workflow ID");
+        }
+
+        final boolean ackRequested = testData.getEdifact().matches(ACK_REQUESTED_REGEX);
+
+        final OutboundMeshMessage outboundMeshMessage = OutboundMeshMessage.create(recipient,
+            workflowId, testData.getEdifact(), null);
+
+        getLabResultsMeshClient().sendEdifactMessage(outboundMeshMessage);
+
+        for (String expectedMessageBody: testData.getJsonList()) {
+            Message gpOutboundQueueMessage = getGpOutboundQueueMessage();
+            assertThat(gpOutboundQueueMessage).isNotNull();
+
+            String correlationId = gpOutboundQueueMessage.getStringProperty(JmsHeaders.CORRELATION_ID);
+            assertThat(correlationId).isNotEmpty();
+
+            String messageBody = parseTextMessage(gpOutboundQueueMessage);
+
+            JSONAssert.assertEquals(
+                expectedMessageBody,
+                messageBody,
+                new CustomComparator(
+                    JSONCompareMode.STRICT,
+                    new Customization("id", IGNORE),
+                    new Customization("meta.lastUpdated", IGNORE),
+                    new Customization("identifier.value", IGNORE),
+                    new Customization("entry[*].fullUrl", IGNORE),
+                    new Customization("entry[*].resource.subject.reference", IGNORE),
+                    new Customization("entry[*].resource.id", IGNORE)
+                )
+            );
+        }
+
+        if (ackRequested) {
+            assertOutboundNhsAckMessage(workflowId);
+        } else {
+            assertThat(getMeshClient().getInboxMessageIds()).isEmpty();
+        }
 
-        final String content = new String(Files.readAllBytes(getEdifactResource().getFile().toPath()));
+    }
 
-        final OutboundMeshMessage outboundMeshMessage = OutboundMeshMessage.create(RECIPIENT,
-            WorkflowId.PATHOLOGY, content, null);
+    /**
+     * The test EDIFACT message (.edifact.dat file) is sent to the MESH mailbox where the adaptor receives inbound
+     * messages. The test waits for the messages to be processed and confirms that the outbound queue is empty.
+     */
+    @ParameterizedTest(name = "[{index}] - {0}")
+    @ArgumentsSource(FailureArgumentsProvider.class)
+    void testEdifactFailsToBeProcessedAndNothingPushedToGpOutboundQueue(String category, TestData testData) {
```

Comment:
category isn't used in the method, but its value is read by the ParameterizedTest annotation; the {0} means the first parameter in the method.

It might be better renaming it to something else to make it more clear, something like testCategoryName or testGroup possibly?

Reply:
I've called it `testGroupName` for now, happy to change it if you or anyone else would prefer something else!

Reply:
That's fine 😄